src/main/java/Duke.java

```diff
@@ -1,10 +1,263 @@
+import java.io.FileNotFoundException;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
+    private static ArrayList<Task> tasks = new ArrayList<>();
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        loadExistingList();
+        printHello();
+        handleCommand();
+        printGoodbye();
+        updateDataFile();
+    }
+
+    private static void loadExistingList() {
+        //If the folder doesn't exists, create it
+        File folder = new File("data");
+        boolean bool = folder.mkdirs();
+//        if(bool){
+//            System.out.println("Directory created successfully");
+//        }else{
+//            System.out.println("Sorry couldn't create specified directory");
+//        }
+
+        //If the file doesn't exist, create it
+        File f = new File("data/duke.txt");
+        Scanner s = null; // create a Scanner using the File as the source
+        try {
+            s = new Scanner(f);
+            parseTasks(s);
+        } catch (FileNotFoundException e) {
+            //System.out.println("File Not Found");
+            //e.printStackTrace();
+            try {
+                f.createNewFile();
+                //System.out.println("New File");
+            } catch (IOException ioException) {
+                ioException.printStackTrace();
+            }
+        }
+    }
+
+    private static void parseTasks(Scanner s) {
+        while (s.hasNext()) {
+            String currentLine = s.nextLine();
+            String[] data = currentLine.split("\\|");
+
+            for (int i = 0; i < data.length; i++) {
+                data[i] = data[i].trim();
+            }
+
+            if (data[0].equals("T")) {
+                tasks.add(new Todo(data[2]));
+                if (data[1].equals("1")) {
```

Comment:
you might want to add variable names for each of the array elements so that its easier for you in the future.